neuralpp/symbolic/sympy_expression.py

```diff
@@ -365,7 +429,7 @@ def subexpressions(self) -> List[Expression]:
 class SymPyFunctionApplication(SymPyFunctionApplicationInterface):
     def __new__(cls, sympy_object: sympy.Basic, type_dict: Dict[sympy.Basic, ExpressionType]):
         if sympy_object.func == sympy.Piecewise:
-            return SymPyConditionalFunctionApplication(sympy_object, type_dict)
+            return SymPyPiecewise(sympy_object, type_dict)
```

Comment:
`SymPyConditionalFunctionApplication` (if-then-else) coexists with `SymPyPiecewise`, but we don't fold piecewise into conditional anymore.